### ISSUE TITLE
NWB file browsing

### ISSUE THREAD

**Issue by arnodelorme** [sccn/NEMAR-pipeline#82]
Yaroslav's message

just in case, here is my setup in apache to allow access to my files
accessed by some 3rd party services... I don't think it is needed for
neurosift but I could be wrong

 <IfModule mod_headers.c>
   # <FilesMatch "\.(nii|nii.gz)$">
   <FilesMatch "\.*$">
     # To offload rendering etc to bioimagesuiteweb
     SetEnvIf Origin "http(s)?://(www\.)?([datalad.org](http://datalad.org/)|[bioimagesuiteweb.github.io](http://bioimagesuiteweb.github.io/)|[bisweb.yale.edu](http://bisweb.yale.edu/)|[neuroglancer-demo.appspot.com](http://neuroglancer-demo.appspot.com/))$" AccessControlAllowOrigin=$0
     Header add Access-Control-Allow-Origin %{AccessControlAllowOrigin}e env=AccessControlAllowOrigin
     Header merge Vary Origin
   </FilesMatch>
 </IfModule>

**Comment by arnodelorme** [comment]
Example 

https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/f0278ddd-f001-463f-b2ba-cf78ca9316f4/download/&dandisetId=000947&dandisetVersion=draft&tab=neurodata-item:/acquisition/ElectricalSeries|ElectricalSeries&tab-time=0.022059545939383057,0.23014721551441283,undefined](https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/f0278ddd-f001-463f-b2ba-cf78ca9316f4/download/&dandisetId=000947&dandisetVersion=draft&tab=neurodata-item:/acquisition/ElectricalSeries%7CElectricalSeries&tab-time=0.022059545939383057,0.23014721551441283,undefined)

**Comment by arnodelorme** [comment]
@cdesyoun example files are in example in the following folder

```
nemar/arno/nwb_example_files
```

**Comment by cdesyoun** [comment]
@arnodelorme , actually, we have public file access permissions via the download url. You can access the following NWB files that you provided.

- https://nemar-dev.ucsd.edu/dataexplorer/download?filepath=/expanse/projects/nemar/openneuro/nwb_example_files/sub-005_ieeg.nwb
- https://nemar-dev.ucsd.edu/dataexplorer/download?filepath=/expanse/projects/nemar/openneuro/nwb_example_files/sub-01_ses-20140828T132700_ecephys+image.nwb
- https://nemar-dev.ucsd.edu/dataexplorer/download?filepath=/expanse/projects/nemar/openneuro/nwb_example_files/sub-09_ses-20170311T170500_ecephys.nwb

When I tested the following link you suggested and replaced the existing URL with nemar-dev's URL, I encountered a 'Loading' page.

https://neurosift.app/?p=/nwb&url=%22https://nemar-dev.ucsd.edu/dataexplorer/download?filepath=/expanse/projects/nemar/openneuro/nwb_example_files/sample%22&dandisetId=000947&dandisetVersion=draft&tab=neurodata-item:/acquisition/ElectricalSeries|ElectricalSeries&tab-time=0.022059545939383057,0.23014721551441283,undefined](https://neurosift.app/?p=/nwb&url=%22https://nemar-dev.ucsd.edu/dataexplorer/download?filepath=/expanse/projects/nemar/openneuro/nwb_example_files/sample%22&dandisetId=000947&dandisetVersion=draft&tab=neurodata-item:/acquisition/ElectricalSeries|ElectricalSeries&tab-time=0.022059545939383057,0.23014721551441283,undefined)

It seems we need to change the parameters of this URL that apply to this dataset."

Thanks,
Choonhan Youn

**Comment by arnodelorme** [comment]
Yes, @cdesyoun . Thank you for trying. I think the permission you need to set are more than download. It is access by a 3rd party site. These are usually blocked. This is what the code I shared does.

**Comment by cdesyoun** [comment]
Arno, in our case, we don't need the provided configuration snippet since NEMAR files are publicly accessible to any third parties. 
This configuration snippet is intended to set cross-origin resource sharing (CORS) headers for specific third-party services accessing files on the server. It ensures that resources can be requested from another domain, which is necessary for web applications to function correctly when fetching resources from different origins. 
The snippet sets an environment variable, 'AccessControlAllowOrigin', to the value of the Origin header if it matches one of the specified domains. 
However, the neurosift domain is not included, which means only the specified domains can access the resources. 
We should ask the neurosift team how to access NEMAR files in their interface and provide the NEMAR data access URLs.

**Comment by arnodelorme** [comment]
OK, would you mind to email Ben Dichter ***@***.***> and ***@***.*** ***@***.***> telling them that the link loads forever.

**Comment by cdesyoun** [comment]
On Neurosift's GitHub page, I found this issue link: https://github.com/flatironinstitute/neurosift/issues/129. 
I left a comment regarding the use of third-party data in the Neurosift interface.
